internal/provider/providers/porkbun/api.go

```diff
@@ -12,9 +12,20 @@ import (
 	"github.com/qdm12/ddns-updater/internal/provider/errors"
 )
 
+// Leaving unused values commented out to improve resilience to API changes.
+type dnsRecord struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+	// Type     string `json:"type"`
+	Content string `json:"content"`
+	// TTL      string `json:"ttl"`
+	// Priority string `json:"prio"`
+	// Notes    string `json:"notes"`
+}
+
 // See https://porkbun.com/api/json/v3/documentation#DNS%20Retrieve%20Records%20by%20Domain,%20Subdomain%20and%20Type
-func (p *Provider) getRecordIDs(ctx context.Context, client *http.Client, recordType string) (
-	recordIDs []string, err error) {
+func (p *Provider) getRecords(ctx context.Context, client *http.Client, recordType string) (
```

Comment:
Let's wrap the recordType in each error returned by `getRecords`, and remove it from wrappings from callers of this function. Since it's this function handling recordType, it should be the only one wrapping errors with it (this helps a lot as a general rule to write non-repetitive error wrappings).

Also my bad this was not case before!

Reply:
I really like this philosophy. I'm hoping what I changed it to makes sense. In general I don't think I have good error message patterns.

Seems like as a general rule of thumb, the caller will error message with what it was invoking, i..e. update, create, and the function should return any relevant error values that will form the context of the error. So if you pass a value to the function, it's more or less expected that the error returned will contain that relevant info, whereas the error log of the caller will include the context of the function itself to avoid duplicating that across every error log inside the function?

i.e. obviously we could put the context of updating, deleting, and creating as well in each error log, and simply log the error in its entirety in the parent without any additional message changes, but then  you get all those strings duplicated in every message.

Does that sound about right?